app/src/main/kotlin/io/intrepid/skotlinton/base/BaseActivity.kt

```diff
@@ -10,18 +10,24 @@ import io.intrepid.skotlinton.SkotlintonApplication
 import timber.log.Timber
 import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper
 
-abstract class BaseActivity : AppCompatActivity() {
+@Suppress("AddVarianceModifier")
+abstract class BaseActivity<S : BaseScreen, P : BasePresenter<S>> : AppCompatActivity() {
 
     protected abstract val layoutResourceId: Int
     protected val skotlintonApplication: SkotlintonApplication
         get() = application as SkotlintonApplication
 
+    protected lateinit var presenter: P
```

Comment:
I'd much rather have presenters be `lazy` than `lateinit`. Does that mess with any of the lifecycle stuff since you don't necessarily know exactly when initialization happens?

Reply:
Since the second activity(fragment) doesn't invoke the presenter, it seemed like a lazy delegate would never call createPresenter(), and the screen would be blank.